system/lib/libc/sbrk.c

```diff
@@ -93,6 +98,25 @@ void *sbrk(intptr_t increment_) {
   }
 }
 
+void *sbrk(intptr_t increment_) {
+#if defined(__wasm64__) // TODO || !defined(wasm2gb)
+  // In the correct https://linux.die.net/man/2/sbrk spec, sbrk() parameter is
+  // intended to be treated as signed, meaning that it is not possible in a
+  // 32-bit program to sbrk alloc (or dealloc) more than 2GB of memory at once.
```

Comment:
If this is indeed the case (I have no reason to believe its not) wouldn't it be better to fix the below bug rather than do this dance?

Maybe at least open a bug so that we can remove this code since it seems like its only need to workaround the bug?

Reply:
https://github.com/emscripten-core/emscripten/issues/25138

Reply:
check